src/GUI/GraphPanel.java

```diff
@@ -45,77 +51,67 @@ public void mouseMoved(MouseEvent e){
         }
     }
 
-    private class CalculateActionListener implements ActionListener {
-        @Override
-        public void actionPerformed(ActionEvent e) {
-            if(functions.size() < 3){
-                String functionString = functionField.getText();
-                double minT = toWorldCoordinates(new Vector2D(0, 0)).x;
-                double maxT = toWorldCoordinates(new Vector2D(getWidth(), 0)).x;
-                PolynomialFunction function = new PolynomialFunction(functionString);
-                function.calcRoots(minT, maxT, 0.001);
-                function.calcExtremePoints(minT, maxT, 0.001);
-                functions.add(function);
-                repaint();
-            } else {
-                GraphPanel.infoBox("You reached the maximum amount of Graphs", "MAX_GRAPHS_REACHED");
-            }
-
+    public int addFunction(String functionString, Color currentColor){
```

Comment:
maybe a comment is warranted here for documenting what is returned. Just from the name, I wouldn't know what the integer, returned by 'addFunction' means.

Reply:
Done